packages/store/src/patterns/patternMatchers.js

```diff
@@ -1207,15 +1485,15 @@ const makePatternKit = () => {
     kind: makeKindMatcher,
     boolean: () => BooleanShape,
     number: () => NumberShape,
-    bigint: () => BigintShape,
-    nat: () => NatShape,
-    string: () => StringShape,
-    symbol: () => SymbolShape,
-    record: () => RecordShape,
-    array: () => ArrayShape,
-    set: () => SetShape,
-    bag: () => BagShape,
-    map: () => MapShape,
+    bigint: (limits = undefined) => makeLimitsMatcher('match:bigint', [limits]),
+    nat: (limits = undefined) => makeLimitsMatcher('match:nat', [limits]),
+    string: (limits = undefined) => makeLimitsMatcher('match:string', [limits]),
+    symbol: (limits = undefined) => makeLimitsMatcher('match:symbol', [limits]),
+    record: (limits = undefined) => M.recordOf(M.any(), M.any(), limits),
+    array: (limits = undefined) => M.arrayOf(M.any(), limits),
+    set: (limits = undefined) => M.setOf(M.any(), limits),
```

Comment:
this means it will do lots of empty matching of "any" on each element.

Reply:
Please add a matcher for just set/bag/etc. without enumerating the elements.

Reply:
Did the next best thing: For `M.arrayOf`, `M.setOf`, and `M.matchOf` if the patterns are all `M.any()` then I skip the enumeration.

For `M.recordOf` and `M.bagOf` I still have to enumerate to enforce the per-element size limit.